src/intersections/segments.jl

```diff
@@ -167,3 +167,86 @@ function intersectcolinear(s1::Segment{Dim,T}, s2::Segment{Dim,T}) where {Dim,T}
     NoIntersection()
   end
 end
+
+# intersection of Segment and Triangle
+function intersecttype(s::Segment{3,T}, t::Triangle{3,T}) where {T}
+  # get the triangle's and segment's vertices
+  s_v = s.vertices
+  t_v = t.vertices
```

Comment:
Notice that whenever we are using these objects in external functions that are not defined in the same source file of the object we can't use internal fields. We need to call the public API `vertices(s)` instead of `s.vertices`.